Failed pipe syntax "from" with URL, method, and headers

```
$ super -c 'from http://ifconfig.co ( method GET headers {Accept:"application/json"} )'
panic: interface conversion: interface {} is semantic.argExpr, not *ast.ArgText

goroutine 1 [running]:
github.com/brimdata/super/compiler/semantic.(*analyzer).exprArg(0xc00016f838, 0xc0001267e0?, {0x117cf2a9, 0x7})
	/Users/phil/work/super/compiler/semantic/op.go:1344 +0x126
github.com/brimdata/super/compiler/semantic.(*analyzer).semFromURL(0xc00016f838, {0x11f03458, 0xc00058ea80}, {0xc000418c18, 0x12}, {0xc0001267e0, 0x2, 0x2})
	/Users/phil/work/super/compiler/semantic/op.go:297 +0x31a
github.com/brimdata/super/compiler/semantic.(*analyzer).semFromName(0xc00016f838, {0x11f03458, 0xc00058ea80}, {0xc000418c18, 0x12}, {0xc0001267e0, 0x2, 0x2})
	/Users/phil/work/super/compiler/semantic/op.go:236 +0xc5
github.com/brimdata/super/compiler/semantic.(*analyzer).semFromEntity(0xc00016f838, {0x11f0ba00?, 0xc00058ea80}, 0x0, {0xc0001267e0, 0x2, 0x2}, {0x0, 0x0, 0x0})
	/Users/phil/work/super/compiler/semantic/op.go:102 +0x5b4
github.com/brimdata/super/compiler/semantic.(*analyzer).semFromElem(0xc00016f838, 0xc0005b0ea0, {0x0?, 0x2?, 0x2?})
	/Users/phil/work/super/compiler/semantic/op.go:57 +0x5b
github.com/brimdata/super/compiler/semantic.(*analyzer).semFrom(0x0?, 0xc000795b78?, {0x0?, 0x1?, 0x0?})
	/Users/phil/work/super/compiler/semantic/op.go:47 +0x3b
github.com/brimdata/super/compiler/semantic.(*analyzer).semOp(0xc00016f838, {0x11f0bb20?, 0xc00033a740}, {0x0, 0x0, 0x0})
	/Users/phil/work/super/compiler/semantic/op.go:559 +0x1b66
github.com/brimdata/super/compiler/semantic.(*analyzer).semSeq(0xc00016f838, {0xc00012a540, 0x1, 0x1})
	/Users/phil/work/super/compiler/semantic/op.go:37 +0x51
github.com/brimdata/super/compiler/semantic.Analyze({0x11f137e0, 0xc000129020}, 0xc000126880, 0xc000126920, 0x0)
	/Users/phil/work/super/compiler/semantic/analyzer.go:25 +0x325
github.com/brimdata/super/compiler.Analyze(...)
	/Users/phil/work/super/compiler/package.go:27
github.com/brimdata/super/compiler.CompileWithAST(0xc000129020, 0x100eedc5?, 0xc000126920, 0x1, 0xc, {0x0, 0x0, 0x0})
	/Users/phil/work/super/compiler/package.go:74 +0x65
github.com/brimdata/super/compiler.(*compiler).NewQuery(0x11f13888?, 0xc00042e000?, 0xc00016f9a8?, {0x0?, 0x8?, 0x11cc3500?}, 0x1?)
	/Users/phil/work/super/compiler/compiler.go:42 +0x45
github.com/brimdata/super/runtime.CompileQuery({0x11f13888?, 0xc00042e000?}, 0xc00042c750, {0x11f025d8, 0xc000088668}, 0xc000126880, {0x0, 0x0, 0x0})
	/Users/phil/work/super/runtime/compiler.go:34 +0x10f
github.com/brimdata/super/cmd/super/root.(*Command).Run(0xc0007a2008, {0xc00003c8f0, 0x0, 0x0})
	/Users/phil/work/super/cmd/super/root/command.go:158 +0x3af
github.com/brimdata/super/pkg/charm.path.run({0xc00042a018, 0x1, 0x1}, {0xc00003c8f0, 0x0, 0x0?})
	/Users/phil/work/super/pkg/charm/path.go:11 +0x7b
github.com/brimdata/super/pkg/charm.(*Spec).Exec(0x13054620, {0xc00003c8e0, 0x2, 0x2})
	/Users/phil/work/super/pkg/charm/charm.go:74 +0x1fa
main.main()
	/Users/phil/work/super/cmd/super/main.go:39 +0x5b
```

## Details

Repro is with super commit 6327e15.

I've been doing more testing on `from` as its [newer docs](https://superdb.org/super-sql/operators/from.html) have been under revision.

I start with a working `curl`.

```
$ curl -X GET -H "Accept: application/json" http://ifconfig.co
{
  "ip": "199.83.220.79",
...
}
```

I can make `super` work if I leave off the header portion.

```
$ super -version
Version: 6327e15ea

$ super -c 'from http://ifconfig.co ( method GET )'
73.162.17.253
```

However, once I tack on the `headers` with a record expression to define the names and values to be included as HTTP header options, I get the stack trace shown above.

--- a/book/src/super-sql/operators/from.md
+++ b/book/src/super-sql/operators/from.md
@@ -135,6 +135,9 @@ to be included as the body of the HTTP request.
 
 Currently, the headers expression must evaluate to a compile-time constant though this
 may change to allow dynamic computation in a future version of SuperSQL.
+Each field of this record must either be a string or (to specify a
+header option appearing multiple times with different values)
+an array or set of strings.
 
 #### Expression
 

--- a/compiler/semantic/op.go
+++ b/compiler/semantic/op.go
@@ -368,20 +368,35 @@ func unmarshalHeaders(val super.Value) (map[string][]string, error) {
 	}
 	headers := map[string][]string{}
 	for i, f := range val.Fields() {
-		if inner := super.InnerType(f.Type); inner == nil || inner.ID() != super.IDString {
-			return nil, errors.New("headers field value must be an array or set of strings")
-		}
 		fieldVal := val.DerefByColumn(i)
 		if fieldVal == nil {
 			continue
 		}
-		for it := fieldVal.Iter(); !it.Done(); {
-			if b := it.Next(); b != nil {
-				headers[f.Name] = append(headers[f.Name], super.DecodeString(b))
-			}
+		headerStrings, err := decodeStrings(fieldVal)
+		if err != nil {
+			return nil, err
 		}
+		headers[f.Name] = append(headers[f.Name], headerStrings...)
 	}
 	return headers, nil
+}
+
+func decodeStrings(val *super.Value) ([]string, error) {
+	typ := super.TypeUnder(val.Type())
+	if inner := super.InnerType(typ); inner != nil {
+		if inner.ID() != super.IDString {
+			return nil, errors.New("array elements of header field must be strings")
+		}
+		var out []string
+		for it := val.Iter(); !it.Done(); {
+			out = append(out, super.DecodeString(it.Next()))
+		}
+		return out, nil
+	}
+	if typ != super.TypeString {
+		return nil, errors.New("header field value must be a string or an array or set of strings")
+	}
+	return []string{val.AsString()}, nil
 }
 
 func (t *translator) fromPoolRegexp(node ast.Node, re, orig, which string, args []ast.OpArg) sem.Seq {
@@ -1651,7 +1666,7 @@ func (t *translator) textArg(o opArgs, key string) (string, ast.Loc) {
 
 func (t *translator) exprArg(o opArgs, key string) (sem.Expr, ast.Loc) {
 	if v, ok := o[key]; ok {
-		if arg, ok := v.(*argExpr); ok {
+		if arg, ok := v.(argExpr); ok {
 			return arg.expr, arg.arg.Loc
 		}
 		// The PEG parser currently doesn't allow this but this may change.

